project/cmake/modules/FindFFMPEG.cmake

```diff
@@ -1,18 +1,140 @@
+  find_library(FFMPEG_LIBAVFORMAT NAMES avformat libavformat PATH_SUFFIXES ffmpeg/libavformat)
+
+  find_library(FFMPEG_LIBAVUTIL
+               NAMES avutil libavutil
+               PATH_SUFFIXES ffmpeg/libavutil
+               PATHS ${PC_FFMPEG_libavutil_LIBDIR}
+               NO_DEFAULT_PATH)
+  find_library(FFMPEG_LIBAVUTIL NAMES avutil libavutil PATH_SUFFIXES ffmpeg/libavutil)
+
+  find_library(FFMPEG_LIBSWSCALE
+               NAMES swscale libswscale
+               PATH_SUFFIXES ffmpeg/libswscale
+               PATHS ${PC_FFMPEG_libswscale_LIBDIR}
+               NO_DEFAULT_PATH)
+  find_library(FFMPEG_LIBSWSCALE NAMES swscale libswscale PATH_SUFFIXES ffmpeg/libswscale)
+
+  find_library(FFMPEG_LIBSWRESAMPLE
+               NAMES swresample libswresample
+               PATH_SUFFIXES ffmpeg/libswresample
+               PATHS ${PC_FFMPEG_libswresample_LIBDIR}
+               NO_DEFAULT_PATH)
+  find_library(FFMPEG_LIBSWRESAMPLE NAMES NAMES swresample libswresample PATH_SUFFIXES ffmpeg/libswresample)
+
+  find_library(FFMPEG_LIBPOSTPROC
+               NAMES postproc libpostproc
+               PATH_SUFFIXES ffmpeg/libpostproc
+               PATHS ${PC_FFMPEG_libpostproc_LIBDIR}
+               NO_DEFAULT_PATH)
+  find_library(FFMPEG_LIBPOSTPROC NAMES postproc libpostproc PATH_SUFFIXES ffmpeg/libpostproc)
+
+  if(PC_FFMPEG_FOUND
+     AND PC_FFMPEG_libavcodec_VERSION
+     AND PC_FFMPEG_libavfilter_VERSION
+     AND PC_FFMPEG_libavformat_VERSION
+     AND PC_FFMPEG_libavutil_VERSION
+     AND PC_FFMPEG_libswscale_VERSION
+     AND PC_FFMPEG_libswresample_VERSION
+     AND PC_FFMPEG_libpostproc_VERSION)
+    set(FFMPEG_VERSION 3.1)
   endif()
 
+
+  include(FindPackageHandleStandardArgs)
+  find_package_handle_standard_args(FFMPEG
+                                    VERSION_VAR FFMPEG_VERSION
+                                    REQUIRED_VARS FFMPEG_INCLUDE_DIRS
+                                                  FFMPEG_LIBAVCODEC
+                                                  FFMPEG_LIBAVFILTER
+                                                  FFMPEG_LIBAVFORMAT
+                                                  FFMPEG_LIBAVUTIL
+                                                  FFMPEG_LIBSWSCALE
+                                                  FFMPEG_LIBSWRESAMPLE
+                                                  FFMPEG_LIBPOSTPROC
+                                                  FFMPEG_VERSION
+                                    FAIL_MESSAGE "FFmpeg 3.1 not found, please consider using -DENABLE_INTERNAL_FFMPEG=ON")
+
+  if(FFMPEG_FOUND)
+    set(FFMPEG_LIBRARIES ${FFMPEG_LIBAVCODEC} ${FFMPEG_LIBAVFILTER}
+                         ${FFMPEG_LIBAVFORMAT} ${FFMPEG_LIBAVUTIL}
+                         ${FFMPEG_LIBSWSCALE} ${FFMPEG_LIBSWRESAMPLE}
+                         ${FFMPEG_LIBPOSTPROC})
+    set(FFMPEG_LDFLAGS ${PC_FFMPEG_LDFLAGS})
+    list(APPEND FFMPEG_DEFINITIONS -DFFMPEG_VER_SHA=\"${FFMPEG_VERSION}\")
```

Comment:
are you sure that quotes need to be escaped here? I've done a similar change for lirc recently and there it was not needed